java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorFileWritingStrategyIT.java

```diff
@@ -247,6 +250,61 @@ public void shouldWriteOneFileInEachLeafPartition() throws Exception {
                     hadoopConfiguration
             );
         }
+
+        @Test
+        public void shouldWriteRecordsWhenThereAreMoreRecordsThanCanFitInLocalStore() throws Exception {
+            // Given
+            RecordGenerator.RecordListAndSchema recordListAndSchema = generateStringRecords("%09d-%s", range(0, 20));
+            StateStore stateStore = createStateStore(recordListAndSchema.sleeperSchema);
+            PartitionTree tree = new PartitionsBuilder(recordListAndSchema.sleeperSchema)
+                    .rootFirst("root")
+                    .splitToNewChildren("root", "L", "R", "000000010")
+                    .buildTree();
+            stateStore.initialise(tree.getAllPartitions());
+            stateStore.fixTime(stateStoreUpdateTime);
+            String ingestLocalWorkingDirectory = createTempDirectory(temporaryFolder, null).toString() + "/path/to/new/sub/directory";
+            IngestCoordinatorTestParameters parameters = createTestParameterBuilder()
+                    .fileNames(List.of("leftFile1", "rightFile1", "leftFile2", "rightFile2"))
+                    .stateStore(stateStore)
+                    .schema(recordListAndSchema.sleeperSchema)
+                    .workingDir(ingestLocalWorkingDirectory)
+                    .build();
+
+            // When
+            ingestRecords(recordListAndSchema, parameters, 5, 10L);
```

Comment:
Can we add some names here to show what these numbers mean? This is similar in the other tests.